src/components/choose-locale/choose-locale.component.test.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { render, fireEvent, wait } from "@testing-library/react";
+import { ChangeLocale } from "./change-locale.component";
+import * as backendController from "./change-locale.resource";
+
+describe(`<ChangeLocale />`, () => {
+  const setupLocaleSelect = async () => {
+    const wrapper = render(
+      <ChangeLocale
+        allowedLocales={["en", "fr", "it", "pt"]}
+        user={{
+          uuid: "uuid",
+          userProperties: {
+            defaultLocale: "en"
+          }
+        }}
+      />
+    );
+    await wait();
+
+    return wrapper.getByDisplayValue("en") as HTMLInputElement;
+  };
+
+  it("should have user's defaultLocale as initial value", async () => {
+    expect((await setupLocaleSelect()).value).toBe("en");
```

Comment:
This test would be a little bit stronger if the defaultLocale was not `en` (e.g. `fr`)